### PR TITLE
Add PCV_SAFE_MOVER_ROLE

### PR DESCRIPTION
**Pull request by thomas-waite**
# Summary
Deploys a new PCVGuardian with broken out roles. Migrates the PCVGuardian and deprecates the old

### CODE REVIEW

File: contracts/pcv/PCVGuardian.sol [hunk 1]
```diff
@@ -81,7 +82,7 @@ contract PCVGuardian is IPCVGuardian, CoreRef {
         uint256 amount,
         bool pauseAfter,
         bool depositAfter
-    ) external override isGovernorOrGuardianOrAdmin {
```
Comment (thomas-waite) [inline contracts/pcv/PCVGuardian.sol:84]: Admin will no longer be able to call the `withdraw()` functions - need to grant whichever addresses had this the PCV_SAFE_MOVER_ROLE if appropriate

File: test/unit/pcv/PCVGuardian.test.ts [hunk 2]
```diff
@@ -209,6 +209,13 @@ describe('PCV Guardian', function () {
       expect(await token.balanceOf(userAddress)).to.eq(1);
     });
 
+    it('should withdraw from a token-pcv deposit when called by the governor', async () => {
```
Comment (thomas-waite) [inline test/unit/pcv/PCVGuardian.test.ts:212]: Sanity check that the permissioning change didn't break anything

File: proposals/description/fip_82b.ts [hunk 3]
```diff
@@ -406,6 +416,26 @@ const fip_82b: ProposalDescription = {
...
+      ],
+      description: 'Grant tribalCouncilTimelock the PCV_SAFE_MOVER_ROLE role'
+    },
+    {
+      target: 'core',
+      values: '0',
+      method: 'grantRole(bytes32,address)',
+      arguments: [
+        '0x6c9ecf07a5886fd74a8d32f4d3c317a7d5e5b5c7a073a3ab06c217e9ce5288e3', // PCV_SAFE_MOVER_ROLE
+        '{optimisticTimelock}'
```
Comment (thomas-waite) [inline proposals/description/fip_82b.ts:436]: Backwards compatibility, TribalCouncil can later revoke when we deprecate the old OA system

File: contracts/core/TribeRoles.sol [hunk 4]
```diff
@@ -101,6 +101,6 @@ library TribeRoles {
     /// @notice capable of changing PCV Deposit and Global Rate Limited Minter in the PSM
     bytes32 internal constant PSM_ADMIN_ROLE = keccak256("PSM_ADMIN_ROLE");
 
-    /// @notice capable of moving PCV on the PCVGuardian
-    bytes32 internal constant PCV_MOVER = keccak256("PCV_MOVER");
+    /// @notice capable of moving PCV between safe addresses on the PCVGuardian
+    bytes32 internal constant PCV_SAFE_MOVER_ROLE = keccak256("PCV_SAFE_MOVER_ROLE");
```
Comment (Joeysantoro) [inline contracts/core/TribeRoles.sol:105]: 👍

File: contracts/pcv/PCVGuardian.sol (hunk 1 above)
Comment (Joeysantoro) [reply contracts/pcv/PCVGuardian.sol:84]: The "admin" should not be used at all going forward. All new contract deployments should explicitly enumerate the roles. I think the above methods should also get new modifiers

File: proposals/description/fip_82b.ts (hunk 3 above)
Comment (Joeysantoro) [reply proposals/description/fip_82b.ts:436]: Does optimistic timelock have Guardian or PCV_ADMIN? If not I prefer removing this

Comment (Joeysantoro) [reply proposals/description/fip_82b.ts:436]: Also, we'd need to actually deploy a new PCVGuardian and transfer the roles over

Comment (thomas-waite) [reply proposals/description/fip_82b.ts:436]: Removing and deploying a new PCVGuardian with migrated state

File: proposals/dao/fip_82b.ts [hunk 5]
```diff
@@ -27,8 +28,16 @@ const fipNumber = 'fip_82b';
 // This should exclusively include new contract deployments
 const deploy: DeployUpgradeFunc = async (deployAddress: string, addresses: NamedAddresses, logging: boolean) => {
   console.log(`No deploy actions for fip${fipNumber}`);
+  const deploySigner = (await ethers.getSigners())[0];
+  const previousPCVGuardian = await ethers.getContractAt('PCVGuardian', addresses.pcvGuardian, deploySigner);
+  const safeAddresses = await previousPCVGuardian.getSafeAddresses();
+
+  const pcvGuardianFactory = await ethers.getContractFactory('PCVGuardian');
+  const newPCVGuardian = await pcvGuardianFactory.deploy(addresses.core, safeAddresses);
```
Comment (Joeysantoro) [inline proposals/dao/fip_82b.ts]: nice

File: contracts/pcv/PCVGuardian.sol (hunk 1 above)
Comment (thomas-waite) [reply contracts/pcv/PCVGuardian.sol:84]: Done

**Review by ditchfieldcaleb — APPROVED** [review]

File: contracts/core/TribeRoles.sol [hunk 6]
```diff
@@ -101,6 +101,9 @@ library TribeRoles {
...
+
+    /// @notice capable of setting and unsetting safe addresses on the PCVGuardian
+    bytes32 internal constant PCV_SAFE_ADMIN_ROLE = keccak256("PCV_SAFE_ADMIN_ROLE");
```
Comment (Joeysantoro) [inline contracts/core/TribeRoles.sol:108]: Prefer to use PCV_GUARDIAN_ADMIN preexisting role

Comment (thomas-waite) [reply contracts/core/TribeRoles.sol:108]: done

**Pull request merged**